frontend/src/app/main/data/workspace.cljs

```diff
@@ -1588,6 +1601,163 @@
              (dwu/commit-undo-transaction))))))))
 
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Remove graphics
+;; TODO: this should be deprecated and removed together with components-v2
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn- initialize-remove-graphics
+  [total]
+  (ptk/reify ::initialize-remove-graphics
+    ptk/UpdateEvent
+    (update [_ state]
+      (assoc state :remove-graphics {:total total
+                                     :current nil}))))
+
+(defn- update-remove-graphics
+  [current]
+  (ptk/reify ::update-remove-graphics
+    ptk/UpdateEvent
+    (update [_ state]
+      (assoc-in state [:remove-graphics :current] current))))
+
+(defn- complete-remove-graphics
+  []
+  (ptk/reify ::complete-remove-graphics
+    ptk/UpdateEvent
+    (update [_ state]
+      (dissoc state :remove-graphics))))
+
+(defn- create-shapes-svg
+  [file-id objects pos media-obj]
+  (let [path (cfg/resolve-file-media media-obj)
+
+        upload-images
+        (fn [svg-data]
+          (->> (svg/upload-images svg-data file-id)
+               (rx/map #(assoc svg-data :image-data %))))
+
+        process-svg
+        (fn [svg-data]
+          (let [[shape children]
+                (svg/create-svg-shapes svg-data pos objects uuid/zero #{} false)]
+          [shape children]))]
+
+    (->> (http/send! {:method :get :uri path :mode :no-cors})
+         (rx/map :body)
+         (rx/map #(vector (:name media-obj) %))
+         (rx/merge-map dwm/svg->clj)
+         (rx/merge-map upload-images)
+         (rx/map process-svg)
+         (rx/catch #(js/console.error  ; When error downloading media-obj, skip it and continue with next one
+                      (str "Error downloading " (:name media-obj) " " path ":")
+                      (clj->js %))))))
+
+(defn- create-shapes-img
+  [pos media-obj]
+  (let [{:keys [name width height id mtype]} media-obj
```

Comment:
the media-obj can be destructured in the function arguments

Reply:
ok